android/src/main/java/com/openxchange/deltachatcore/NativeInteractionManager.java

```diff
@@ -329,107 +317,34 @@ public long handleEvent(final int event, long data1, long data2) {
-                    try {
-                        conn.setConnectTimeout(15 * 1000);
-                        conn.setReadTimeout(15 * 1000);
-                        conn.setDoOutput(true);
-                        conn.setRequestMethod("POST");
-                        conn.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
-                        conn.setRequestProperty("Content-Length", String.valueOf(bytes.length));
-                        conn.getOutputStream().write(bytes);
-
-                        int responseCode = conn.getResponseCode();
-                        BufferedReader br = new BufferedReader(new InputStreamReader(new BufferedInputStream(conn.getInputStream())));
-                        StringBuilder total = new StringBuilder();
-                        String line;
-                        while ((line = br.readLine()) != null) {
-                            total.append(line).append('\n');
-                        }
-                        if (responseCode == HttpURLConnection.HTTP_OK) {
-                            postContent = total.toString();
-                        } else {
-                            Log.i(TAG, String.format("DC_EVENT_HTTP_POST error: %s", total.toString()));
-                        }
-                    } finally {
-                        conn.disconnect();
-                    }
-                } catch (Exception e) {
-                    e.printStackTrace();
-                }
-                return stringToData(postContent);
-
-            case DC_EVENT_GET_STRING:
-                String coreString = null;
-                if (coreStrings != null && !coreStrings.isEmpty()) {
-                    coreString = coreStrings.get(data1);
-                }
-                return stringToData(coreString);
-
-            default: {
-                final Object data1obj = data1IsString(event) ? dataToString(data1) : data1;
-                final Object data2obj = data2IsString(event) ? dataToString(data2) : data2;
-                if (eventChannelHandler != null) {
-                    eventChannelHandler.handleEvent(event, data1obj, data2obj);
-                }
-            }
-            break;
         }
         return 0;
     }
 
-    private void handleError(int event, String error) {
-        eventChannelHandler.handleEvent(EVENT_ERROR, event, error);
+    private void delegateEvent(int eventId, long data1, long data2) {
+        final Object data1Object = getDataObject(eventId, data1);
+        final Object data2Object = getDataObject(eventId, data2);
+        if (eventChannelHandler != null) {
+            eventChannelHandler.handleEvent(eventId, data1Object, data2Object);
+        }
+    }
+
+    private Object getDataObject(int eventId, long data) {
+        return data1IsString(eventId) ? dataToString(data) : data;
+    }
+
+    private void delegateError(int eventId, long data1, String error) {
```

Comment:
Intentional logic change. We now delegate the event id, data1 and the data2 field as string to the Flutter part. Before we delegated "400" (error event id), event id and data2 as string.

Reply:
I guess I have to implement it the same way on iOS.

Reply:
Makes sense, we should align the native platforms.